### PR TITLE
feat: Status notifications

### PR DESCRIPTION
**Pull request by makingthematrix**
The status (availability) affects the notifications now. If the status is set to 'Away' the notifications are not displayed at all. If it's 'Busy' the notifications are displayed only if certain requirements are met. Calls also are affected in a similar way.

I also made a small change to `app.gradle`, because otherwise I was not able to use SE.

SE part: https://github.com/wireapp/wire-android-sync-engine/pull/538
#### APK
[Download build #12596](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12596/artifact/build/artifact/wire-dev-PR2117-12596.apk)
[Download build #12597](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12597/artifact/build/artifact/wire-dev-PR2117-12597.apk)
[Download build #12598](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12598/artifact/build/artifact/wire-dev-PR2117-12598.apk)
[Download build #12599](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12599/artifact/build/artifact/wire-dev-PR2117-12599.apk)
[Download build #12600](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12600/artifact/build/artifact/wire-dev-PR2117-12600.apk)
[Download build #12601](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12601/artifact/build/artifact/wire-dev-PR2117-12601.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/notifications/controllers/CallingNotificationsController.scala [hunk 1]
```diff
@@ -139,27 +155,28 @@ class CallingNotificationsController(implicit cxt: WireContext, eventContext: Ev
...
-          try showNotification()
-          catch {
-            case NonFatal(e2) => error(l"second display attempt failed, aborting", e2)
-          }
-      }
+        Try(showNotification()).recover {
+          case NonFatal(e) =>
+            error(l"Notify failed: try without bitmap", e)
+            builder.setLargeIcon(null)
+            try showNotification()
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/notifications/controllers/CallingNotificationsController.scala]: Select please one style. `Try` or `try/catch`. Not both of them in one place.

File: app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala [hunk 2]
```diff
@@ -195,6 +220,34 @@ class MainPhoneFragment extends FragmentHelper
...
+          case Availability.BUSY      => (R.string.availability_notification_warning_busy_title,      R.string.availability_notification_warning_busy)
+          case Availability.AVAILABLE => (R.string.availability_notification_warning_available_title, R.string.availability_notification_warning_available)
+          case Availability.NONE      => (R.string.availability_notification_warning_nostatus_title,  R.string.availability_notification_warning_nostatus)
+        }
+        showConfirmationDialog(
+          getString(title),
+          getString(body),
+          R.string.availability_notification_dont_show,
+          R.string.availability_notification_ok
+        ).map {
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala]: We do not need pattern matching here. `filter.foreach` will do the job.

File: app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala [hunk 3]
```diff
@@ -91,44 +92,69 @@ class MainPhoneFragment extends FragmentHelper
...
-      ).flatMap { resp =>
-        zms.head.flatMap { zms =>
-          for {
-            _ <- zms.userPrefs(CrashesAndAnalyticsRequestShown) := true
-            _ <- zms.prefs(analyticsPrefKey) := resp //we override whatever the global value is on asking the user again
-            _ <- if (resp) inject[GlobalTrackingController].optIn() else Future.successful(())
-          } yield {}
-        }
-      }
+  private lazy val consentDialog = for {
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala]: Looks too complicated... Maybe `async/await` can help?

Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala]: I only changed indentation to make the code more readable. I don't want to make any serious changes to it in this PR.

File: app/src/main/scala/com/waz/zclient/notifications/controllers/CallingNotificationsController.scala (hunk 1 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/notifications/controllers/CallingNotificationsController.scala]: ok

File: app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala (hunk 2 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/pages/main/MainPhoneFragment.scala]: ok

**Pull request merged**